source/dub/packagesupplier.d

```diff
@@ -191,7 +191,17 @@ class RegistryPackageSupplier : PackageSupplier {
 		auto vers = best["version"].get!string;
 		auto url = m_registryUrl ~ Path(PackagesPath~"/"~packageId~"/"~vers~".zip");
 		logDiagnostic("Downloading from '%s'", url);
-		download(url, path);
+		foreach(i; 0..3) {
+			try{
+				download(url, path);
+				break;
+			}
+			catch(HTTPStatusException e) {
+				logDebug("Failed to download package %s from %s (Attempt %s of 3)", packageId, url, i + 1);
```

Comment:
There should be a check for 404 errors here, or maybe even limit the retry behavior to 500 errors and connection failures. Retrying for 404 errors would mean that package download in scenarios with multiple registries (that contain different sets of packages) potentially gets slowed down considerably.